src/components/elements/Map-mapbox/MapControls/CheckPolygonControl.tsx

```diff
@@ -0,0 +1,141 @@
+import { SitePolygon } from "@/generated/apiSchemas";
+
+import Button from "../../Button/Button";
+import Text from "../../Text/Text";
+
+export interface CheckSitePolygonProps {
+  siteRecord?: {
+    uuid: string;
+  };
+}
+
+interface CheckedPolygon {
+  uuid: string;
+  valid: boolean;
+  checked: boolean;
+}
+
+interface TransformedData {
+  id: number;
+  valid: boolean;
+  checked: boolean;
+  label: string | null;
+}
+
+const CheckPolygonControl = (props: CheckSitePolygonProps) => {
+  const { siteRecord } = props;
+  const siteUuid = siteRecord?.uuid;
+  const [openCollapse, setOpenCollapse] = useState(false);
+  const [sitePolygonCheckData, setSitePolygonCheckData] = useState<TransformedData[]>([]);
+  const [clickedValidation, setClickedValidation] = useState(false);
+  const context = useSitePolygonData();
+  const sitePolygonData = context?.sitePolygonData;
+  const { data: currentValidationSite, refetch: reloadSitePolygonValidation } = useGetV2TerrafundValidationSite<
+    CheckedPolygon[]
+  >(
+    {
+      queryParams: {
+        uuid: siteUuid ?? ""
+      }
+    },
+    {
+      enabled: !!siteUuid
+    }
+  );
+
+  const validatePolygons = () => {
+    fetchGetV2TerrafundValidationSitePolygons({
+      queryParams: {
+        uuid: siteUuid ?? ""
+      }
+    }).then(() => {
+      reloadSitePolygonValidation();
+      setClickedValidation(false);
+    });
+  };
+
+  const getTransformedData = (currentValidationSite: CheckedPolygon[]) => {
+    return currentValidationSite.map((checkedPolygon, index) => {
+      const matchingPolygon = Array.isArray(sitePolygonData)
+        ? sitePolygonData.find((polygon: SitePolygon) => polygon.poly_id === checkedPolygon.uuid)
+        : null;
+      return {
+        id: index + 1,
+        valid: checkedPolygon.valid,
+        checked: checkedPolygon.checked,
+        label: matchingPolygon?.poly_name ?? null
+      };
+    });
+  };
+
+  useEffect(() => {
+    if (currentValidationSite) {
+      const transformedData = getTransformedData(currentValidationSite);
+      setSitePolygonCheckData(transformedData);
+    }
+  }, [currentValidationSite, sitePolygonData]);
+
+  useEffect(() => {
+    if (sitePolygonData) {
+      reloadSitePolygonValidation();
+    }
+  }, [sitePolygonData]);
+
+  useEffect(() => {
+    if (clickedValidation) {
+      validatePolygons();
+    }
+  }, [clickedValidation]);
+
+  return (
+    <div className="grid gap-2">
+      <div className="rounded-lg bg-[#ffffff26] p-3 text-center text-white backdrop-blur-md">
+        <Text variant="text-10-light">Your polygons have been updated</Text>
+        <Button
+          variant="text"
+          className="text-10-bold my-2 flex w-full justify-center rounded-lg border border-tertiary-600 bg-tertiary-600 p-2 hover:border-white"
+          onClick={() => setClickedValidation(true)}
+        >
+          Check Polygons
```

Comment:
Wrap the free text in this component in `t()` please.

Reply:
done